src/main/scala/devices/tilelink/BusBypass.scala

```diff
@@ -92,7 +92,7 @@ class TLBusBypassBar(dFn: TLSlavePortParameters => TLSlavePortParameters)(implic
     out0.d.ready := in.d.ready &&  bypass
     out1.d.ready := in.d.ready && !bypass
     in.d.valid   := Mux(bypass, out0.d.valid, out1.d.valid)
-    def cast(x: TLBundleD) = { val out = Wire(in.d.bits); out <> x; out }
+    def cast(x: TLBundleD) = { val out = WireDefault(in.d.bits); out <> x; out }
```

Comment:
I don't think this is OK.
`Wire(in.d.bits)` only use the hardware type and generate a `Wire` hardware, but what you are proposing here is giving `out` a non constant value with `in.d.bits`.

Reply:
Chisel.Wire is chisel3.WireDefault:
https://github.com/chipsalliance/chisel3/blob/fc970ca28e562f2ea3ba160963604ea3deaf3467/src/main/scala/chisel3/compatibility.scala#L72-L80

Reply:
oh fk